source/compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -122,6 +122,20 @@ pub enum Error {
     #[error("failed to evaluate: {0} is not supported")]
     #[diagnostic(code("Qsc.PartialEval.Unimplemented"))]
     Unimplemented(String, #[label] PackageSpan),
+
+    #[error("unsupported call into test callable")]
+    #[diagnostic(code("Qsc.PartialEval.UnsupportedTestCallable"))]
+    #[diagnostic(help(
+        "callables with the `@Test` annotation are not supported in partial evaluation."
+    ))]
+    UnsupportedTestCallable(#[label] PackageSpan),
+
+    #[error("unsupported use of simulation-only intrinsic `{0}`")]
+    #[diagnostic(code("Qsc.PartialEval.UnsupportedSimulationIntrinsic"))]
+    #[diagnostic(help(
+        "intrinsic callables that require a running quantum simulator cannot be invoked in partial evaluation"
```

Comment:
Same comment as above. Let's make this error understandable beyond folks that work on our team :-)

Reply:
I'll need a paragraph or two to explain the concepts then... is the request here to improve the help text or the error text? I think the error text is already the concise, easy to understand version: "unsupported use of simuilation only intrinsic `<name>`."